projects/storefrontlib/src/cms-components/navigation/navigation/navigation-ui.component.ts

```diff
@@ -71,6 +89,41 @@ export class NavigationUIComponent implements OnDestroy {
     );
   }
 
+  /**
+   * During initialization of this component, we will check the resetMenuOnClose flag and attach a menu reset listener if needed.
+   */
+  ngOnInit() {
+    if (this.config.resetMenuOnClose) {
+      this.resetOnMenuCollapse();
+    }
+  }
+
+  /**
+   * This method performs the action of resetting the menu (close all sub menus and return to main options)
+   * when the menu is closed.
+   */
+  resetOnMenuCollapse(): void {
+    this.subscriptions.add(
+      this.hamburgerMenuService?.isExpanded
+        .pipe(distinctUntilChanged())
+        .subscribe((isExpanded: boolean) => {
+          if (isExpanded) {
```

Comment:
Why is `isExpanded` not enough? It is possible to be expanded without open nodes? If so, can these conditions be in the same IF statement with `&&`?